### PR TITLE
fix lazy init always "en"

### LINKED ISSUE #87: Now only the first time is i18n
my code:
```rust
rust_i18n::i18n!("locales");

fn main() {
    rust_i18n::set_locale("zh-CN");
    for _ in 0..5 {
        show_message(t!("ok"), t!("hello"));
        thread::sleep(Duration::from_secs(1));
    }
}

fn show_message(title: impl AsRef<str>, message: impl AsRef<str>) {
    rfd::MessageDialog::new()
        .set_title(title.as_ref())
        .set_description(message.as_ref())
        .show();
}
```
zh-CN.yaml
```yaml
ok: "好的"
hello: "你好 世界"
```
en-US.yaml
```yaml
ok: "OK"
hello: "Hello World"
```
***first time***
image
***second time***
image

It can be found that even the first call will not be fully ```i18n```
The second or subsequent calls will never find the yaml content

### PR DESCRIPTION
**Pull request by yk0n9**
close #87

### CODE REVIEW

File: crates/macro/src/lib.rs [hunk 1]
```diff
@@ -292,7 +292,12 @@ fn generate_code(
 
     let default_locale = if let Some(default_locale) = args.default_locale {
         quote! {
-            rust_i18n::set_locale(#default_locale);
+            use std::ops::Deref;
+            if #default_locale != rust_i18n::locale().deref() {
+                rust_i18n::set_locale(rust_i18n::locale().deref());
```
Comment (huacnlee) [inline crates/macro/src/lib.rs:297]: There is not need to set again, just do nothing?

Comment (yk0n9) [reply crates/macro/src/lib.rs:297]: Yes, I have tested it. This will cause it to get the value of ```CURRENT_LOCALE``` when it is init

**Comment by KKRainbow** [comment]
Could you please add me as a co-author?

File: crates/macro/src/lib.rs (hunk 1 above)
Comment (huacnlee) [reply crates/macro/src/lib.rs:297]: I mean this line set_locale is meaningless, the locale is already that value.

**Comment by yk0n9** [comment]
> Could you please add me as a co-author?

of course

File: crates/macro/src/lib.rs (hunk 1 above)
Comment (yk0n9) [reply crates/macro/src/lib.rs:297]: The default value of ```CURRENT_LOCALE``` is "en". These codes will not read the value set by set_locale during ```runtime```. Instead, they will overwrite the set_locale of ```runtime``` to "en" during initialization. The purpose of this modification is to obtain and apply the modified ```CURRENT_LOCALE``` during runtime during initialization.

**Comment by huacnlee** [comment]
Hi @yk0n9 I have make a new change to do this.

https://github.com/longbridgeapp/rust-i18n/pull/90

**Comment by yk0n9** [comment]
> Hi @yk0n9 I have make a new change to do this.
> 
> #90

I feel like your code doesn't fix it😂

1. ```_RUST_I18N_BACKEND``` is lazy, set_locale does not trigger ```_RUST_I18N_BACKEND``` initialization, t!() will trigger ```_RUST_I18N_BACKEND``` initialization.

```rust
set_locale("zh");
t!("xxx"); // <-- backend is initialized here
```
2. The initialization process only has the default value, and will not get the value from ```CURRENT_LOCALE```. This will make the backend always "en" and will overwrite the first line of ```set_locale("zh");```.

3. We must use locale() to get ```CURRENT_LOCALE``` while initializing ```_RUST_I18N_BACKEND```. It ensures that the first line ```set_locale("zh");``` will not be overwritten.

**Comment by yk0n9** [comment]
```rust
    #[test]
    fn test_set_locale() {
        rust_i18n::set_locale("zh-CN");
        for _ in 0..5 {
            assert_eq!(t!("hello"), "Bar - 你好世界！");
        }
    }
```
```rust
---- tests::test_set_locale stdout ----
thread 'tests::test_set_locale' panicked at tests\integration_tests.rs:323:13:
assertion `left == right` failed
  left: "Bar - Hello, World!"
 right: "Bar - 你好世界！"
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library\std\src/panicking.rs:652:5
   1: core::panicking::panic_fmt
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library\core\src/panicking.rs:72:14
   2: core::panicking::assert_failed_inner
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library\core\src/panicking.rs:408:17
   3: core::panicking::assert_failed
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23\library\core\src/panicking.rs:363:5
   4: integration_tests::tests::test_set_locale
             at .\tests\integration_tests.rs:323:13
   5: integration_tests::tests::test_set_locale::{{closure}}
             at .\tests\integration_tests.rs:320:25
   6: core::ops::function::FnOnce::call_once
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23\library\core\src\ops/function.rs:250:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library\core\src\ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    tests::test_set_locale

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 18 filtered out; finished in 0.01s

error: test failed, to rerun pass `-p rust-i18n --test integration_tests`
```
Apparently, the test has failed in the merged code

**Comment by huacnlee** [comment]
Oops, I remember I passed the test. 😂

**Comment by huacnlee** [comment]
Please merge main into your branch, I am going to merge this.

**Pull request merged**